fs/accounting.go

```diff
@@ -394,12 +444,12 @@ func (acc *Account) read(in io.Reader, p []byte) (n int, err error) {
 	// Get the token bucket in use
 	tokenBucketMu.Lock()
 	tb := tokenBucket
-	tokenBucketMu.Unlock()
 
 	// Limit the transfer speed if required
 	if tb != nil {
 		tb.Wait(int64(n))
 	}
+	tokenBucketMu.Unlock()
```

Comment:
The tb.Wait() is thread safe - if you move the lock until after it then you'll lock the tokenBucketMu while it is waiting, which will cause all the other threads to pile up waiting for the lock. So I think this change is incorrect.